src/features/customerView/createLease/useCreateLease.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+
+import CreateLeaseModal from './CreateLeaseModal';
+
+// FIXME: This comment is intended for whoever picks this up next week
+// This approach is an attempt to clear up the amount of state, query and render control in CustomerView,
+// by separating modal features into a hook and a component. Feel free to use this or change it as you see fit!
+// With an approach like this, this whole flow can be attached to any view. Perhaps this could be under 'features'?
```

Comment:
What about this comment?

Reply:
I had one issue with this approach that the data was queried when the render modal method was called; meaning when the customer page was loaded and I couldn't manage to make it fetch its own data separately whenever you open the modal.
This is a problem if you were to create a new boat for example, you had to refresh the while page.
Therefore, I switched the custom hook back to our conventional way.

Reply:
👍  Fair enough, I guess it was worth a try